### ISSUE TITLE
Not a bug but a question: what is port 8988 used for list_queues?

### ISSUE THREAD

**Issue by IronCore864** [rabbitmq/rabbitmq-cli#249]
I'm running a cluster of 3 nodes within AWS with very strict security rules, allowing access to ports that are only described in manual.

Some time later, `rabbitmqctl list_queues` just hanged there and never finished until the default 60s timeout was reached. I used `strace` to figure out what happened, and it seems `recvfrom...` timed out.

After I updated the security group, allowing every port, I tried `strace` again, and it worked, and here're the result:

```
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=11991, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=11992, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=11993, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=11994, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=11995, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=11996, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=11997, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=11998, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
socket(PF_NETLINK, SOCK_RAW, NETLINK_AUDIT) = 3
sendto(3, "p\0\0\0O\4\5\0\1\0\0\0\0\0\0\0op=PAM:setcred a"..., 112, 0, {sa_family=AF_NETLINK, pid=0, groups=00000000}, 12) = 112
recvfrom(3, "$\0\0\0\2\0\0\0\1\0\0\0\326.\0\0\0\0\0\0p\0\0\0O\4\5\0\1\0\0\0"..., 8988, MSG_PEEK|MSG_DONTWAIT, {sa_family=AF_NETLINK, pid=0, groups=00000000}, [12]) = 36
recvfrom(3, "$\0\0\0\2\0\0\0\1\0\0\0\326.\0\0\0\0\0\0p\0\0\0O\4\5\0\1\0\0\0"..., 8988, MSG_DONTWAIT, {sa_family=AF_NETLINK, pid=0, groups=00000000}, [12]) = 36
socket(PF_LOCAL, SOCK_DGRAM|SOCK_CLOEXEC, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/dev/log"}, 110) = 0
sendto(3, "<86>Mar 15 14:29:27 runuser: pam"..., 103, MSG_NOSIGNAL, NULL, 0) = 103
socket(PF_NETLINK, SOCK_RAW, NETLINK_AUDIT) = 4
sendto(4, "x\0\0\0Q\4\5\0\2\0\0\0\0\0\0\0op=PAM:session_o"..., 120, 0, {sa_family=AF_NETLINK, pid=0, groups=00000000}, 12) = 120
recvfrom(4, "$\0\0\0\2\0\0\0\2\0\0\0\326.\0\0\0\0\0\0x\0\0\0Q\4\5\0\2\0\0\0"..., 8988, MSG_PEEK|MSG_DONTWAIT, {sa_family=AF_NETLINK, pid=0, groups=00000000}, [12]) = 36
recvfrom(4, "$\0\0\0\2\0\0\0\2\0\0\0\326.\0\0\0\0\0\0x\0\0\0Q\4\5\0\2\0\0\0"..., 8988, MSG_DONTWAIT, {sa_family=AF_NETLINK, pid=0, groups=00000000}, [12]) = 36
Timeout: 60.0 seconds ...
Listing queues for vhost / ...
sendto(3, "<86>Mar 15 14:29:28 runuser: pam"..., 88, MSG_NOSIGNAL, NULL, 0) = 88
socket(PF_NETLINK, SOCK_RAW, NETLINK_AUDIT) = 4
sendto(4, "x\0\0\0R\4\5\0\3\0\0\0\0\0\0\0op=PAM:session_c"..., 120, 0, {sa_family=AF_NETLINK, pid=0, groups=00000000}, 12) = 120
recvfrom(4, "$\0\0\0\2\0\0\0\3\0\0\0\326.\0\0\0\0\0\0x\0\0\0R\4\5\0\3\0\0\0"..., 8988, MSG_PEEK|MSG_DONTWAIT, {sa_family=AF_NETLINK, pid=0, groups=00000000}, [12]) = 36
recvfrom(4, "$\0\0\0\2\0\0\0\3\0\0\0\326.\0\0\0\0\0\0x\0\0\0R\4\5\0\3\0\0\0"..., 8988, MSG_DONTWAIT, {sa_family=AF_NETLINK, pid=0, groups=00000000}, [12]) = 36
socket(PF_NETLINK, SOCK_RAW, NETLINK_AUDIT) = 4
sendto(4, "p\0\0\0P\4\5\0\4\0\0\0\0\0\0\0op=PAM:setcred a"..., 112, 0, {sa_family=AF_NETLINK, pid=0, groups=00000000}, 12) = 112
recvfrom(4, "$\0\0\0\2\0\0\0\4\0\0\0\326.\0\0\0\0\0\0p\0\0\0P\4\5\0\4\0\0\0"..., 8988, MSG_PEEK|MSG_DONTWAIT, {sa_family=AF_NETLINK, pid=0, groups=00000000}, [12]) = 36
recvfrom(4, "$\0\0\0\2\0\0\0\4\0\0\0\326.\0\0\0\0\0\0p\0\0\0P\4\5\0\4\0\0\0"..., 8988, MSG_DONTWAIT, {sa_family=AF_NETLINK, pid=0, groups=00000000}, [12]) = 36
+++ exited with 0 +++
```

Question is, what is this `recvfrom 8988`? Didn't find any number like this in any config or even source code.

Weird thing is, after it could work, I disabled the security group that allows access to this port, and `list_queues` would always work.

Anyone knows what is it?

Thanks!

It baffles me for days.

**Comment by lukebakken** [comment]
Hello, please post this question to the [mailing list](https://groups.google.com/forum/#!forum/rabbitmq-users). We can answer it there and help out the entire RabbitMQ community. Thank you!

**Comment by michaelklishin** [comment]
See [Port Access](https://www.rabbitmq.com/networking.html#selinux-ports) and [How Nodes and CLI Tools Authenticate to Each Other](http://www.rabbitmq.com/clustering.html#erlang-cookie).

**Comment by michaelklishin** [comment]
Also this [really long rabbitmq-users thread](https://groups.google.com/d/msg/rabbitmq-users/OiiVP_RjP5A/D3vJ1AeGAgAJ) with effectively the same question.

**Comment by IronCore864** [comment]
Thanks for the update. I traced down exactly to the code Luke Bakken pasted in the thread, but I didn't quite understand how a port number is being selected and generated so I gave up and asked here.

I did some test and found out that the default range is still not 35672-35682, although I'm using 3.7.3. But assigning ranges by setting kernel.inet_dist_listen_min and kernel.inet_dist_listen_max it works.
